### PR TITLE
Feature/all pairs

### PR DESCRIPTION
**Pull request by hochbruj**
Draft for selecting V2 pairs instead of tokens -> Not ready to merge! Should be used as input for building new APIs:
- getPairs reads all deployed V2 pairs from chain
- refactored getVolume and getReserves -> currently only working for pairs with WVET token

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -50,132 +53,146 @@ const HeaderWrapper = styled.div`
...
+    if (!vetPrice || tokens.length > 0) return;
+    const promises = allTokens.map(async (item) => {
+      const { data } = await axios.get(`/api/${item.address}`);
+      item = { ...data, ...item };
 
-        // Extra attribute for sorting API
-        // As sorting API cannot handle nested objects
-        item.priceInVet = item.price.base2quote;
+      // Multiplied by two because it's a 50-50 pool
+      item.tvlInUsd = item.reserves * item.price.base2quote * vetPrice * 2;
```
Comment (OliverNChalk) [inline pages/index.js:112]: Is quote guaranteed to be `VET`?

File: utils/api/volume.js [hunk 2]
```diff
@@ -1,58 +1,52 @@
...
+export const getETHPosition = (pair) => {
   const wvet = WVET[1];
-  if (pair.token0.address === wvet.address) {
+  if (pair[0] === wvet.address) {
     return 0;
-  } else {
+  } else if (pair[1] === wvet.address) {
     return 1;
-  }
+  } else return 3;
```
Comment (OliverNChalk) [inline utils/api/volume.js]: Return `-1`, it's more C like?

File: pages/api/[pool].js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ]);
+
+  const vetPosition = getETHPosition(pair);
+
+  let volumeInVet = 0;
+  let reserves = 0;
+  let price = 0;
+
+  //pools without VET are not considered
+  if (vetPosition !== 3) {
```
Comment (OliverNChalk) [inline pages/api/[pool].js]: So this PR is just a refactor because we do not yet support nonVET pairs?

File: components/TokenTable/index.js [hunk 4]
```diff
@@ -174,11 +176,11 @@ const TokenTable = ({ tokens, vetPrice, itemMax = 10, useTracked = false }) => {
                     setSortDirection(sortedColumn !== SORT_FIELD.NAME ? true : !sortDirection)
                   }}
                 >
-                  {ifBelow680 ? 'Symbol' : 'Name'} {sortedColumn === SORT_FIELD.NAME ? (!sortDirection ? '↑' : '↓') : ''}
+                  {'Pair'} {sortedColumn === SORT_FIELD.NAME ? (!sortDirection ? '↑' : '↓') : ''}
                 </ClickableText>
               </Flex>
             </th>
-            {!ifBelow640 && (
```
Comment (KennethAshley) [inline components/TokenTable/index.js]: I think we can remove the name and symbol and do something like VEX-VET, SHA-VET, etc. Similar to https://v2.info.uniswap.org/pairs

File: pages/index.js [hunk 5]
```diff
@@ -60,85 +63,108 @@ const HeaderWrapper = styled.div`
...
-  const [tokens, setTokens] = useState([])
-  const [tvl, setTVL] = useState(0)
-  const [vol, setVol] = useState(0)
+  const [vetPrice, setVetPrice] = useState(0);
+  const [tokens, setTokens] = useState([]);
+  const [pairs, setPairs] = useState([]);
+  const [tvl, setTVL] = useState(0);
+  const [vol, setVol] = useState(0);
+
+  console.log("all pairs ", pairs);
```
Comment (KennethAshley) [inline pages/index.js]: we'd want to remove this

**Review by KennethAshley — CHANGES_REQUESTED** [review]
Just a couple comments but nice work on this Jan!

**Review by xenide — APPROVED** [review]

**Review by OliverNChalk — APPROVED** [review]

**Pull request merged**